exist-core/src/test/xquery/fn.xql

```diff
@@ -126,3 +126,46 @@ function fnt:tokenize-onearg($str as xs:string) {
      tokenize($str)
 };
 
+declare
+    %test:assertError("err:XPDY0002")
+function fnt:document-uri0_noarg() {
+     document-uri()
+};
+
+declare
+    %test:assertError("err:XPDY0002")
+function fnt:document-uri0() {
+     document-uri(.)
+};
+
+declare
+    %test:assertEquals("/db/fn-test/test.xml")
+function fnt:document-uri0_context() {
+     root(collection('/db/fn-test')//book)/document-uri()
+};
+
+declare
+    %test:assertError("err:XPDY0002")
+function fnt:document-uri0_context_empty() {
+     root(collection('/db/fn-test')//bookies)/document-uri()
+};
+
+declare
+    %test:assertEquals("/db/fn-test/test.xml")
+function fnt:document-uri1() {
+     document-uri(root(collection('/db/fn-test')//book))
+};
+
+declare
+    %test:assertEmpty
+function fnt:document-uri1_empty() {
+     document-uri(root(collection('/db/fn-test')//bookies))
+};
+
+declare
+    %test:assertEquals("TRUE")
+function fnt:fn-document-uri-32() {
```

Comment:
In the description you reference fn:document-uri-33 as a possible false negative. Is this the one you added to track that or did I misread your comment?

Reply:
OK, they likely only differ in calling upper-case (33) and lower-case (32), right?

Reply:
indeed....